### PR TITLE
Fix for issue 306

### PR DESCRIPTION
**Pull request by rjnb50**
This is a proof of concept fix for issue 306.

It's a little bit rough and ready, only implemented in GenericFileBrowser and not also in GridViewBrowser and will need a bit of tidying up, but it provides a good and workable fix for the issue.

### CODE REVIEW

**Comment by rjnb50** [comment]
This is issue 306: https://github.com/files-community/files-uwp/issues/306

**Comment by yaichenbaum** [comment]
@rjnb50 The issue is only present in GenericFileBrowser so it shouldn't be necessary to implement it in GridViewBrowser. Let us know if this is ready for review.

File: Files/Views/LayoutModes/GenericFileBrowser.xaml.cs [hunk 1]
```diff
@@ -436,9 +440,26 @@ protected override ListedItem GetItemFromElement(object element)
             return row.DataContext as ListedItem;
         }
 
-        private void RightClickContextMenu_Opening(object sender, object e)
+        [DllImport("user32.dll", CharSet = CharSet.Auto, ExactSpelling = true)]
```
Comment (duke7553) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml.cs:443]: iirc user32.dll is not available to UWP apps.

**Comment by duke7553** [comment]
@rjnb50 Would it be a better idea to set a Task.Delay() here than fiddling with DateTime ?

File: Files/Views/LayoutModes/GenericFileBrowser.xaml.cs (hunk 1 above)
Comment (tsvietOK) [reply Files/Views/LayoutModes/GenericFileBrowser.xaml.cs:443]: It is [available](https://docs.microsoft.com/en-us/uwp/win32-and-com/win32-extension-apis#apis-from-user32dll), but there is no method which is used in this PR.

Comment (duke7553) [reply Files/Views/LayoutModes/GenericFileBrowser.xaml.cs:443]: @tsvietOK You're right, it's just very limited.

Comment (rjnb50) [reply Files/Views/LayoutModes/GenericFileBrowser.xaml.cs:443]: Thank you; you are both correct. I should have used

```
private readonly Windows.UI.ViewManagement.UISettings uiSettings = new Windows.UI.ViewManagement.UISettings();
uiSettings.DoubleClickTime
```

I'm working on a new version of this fix with a couple of other small tweaks to the code and then I'll update the PR in a few days.

Comment (tsvietOK) [reply Files/Views/LayoutModes/GenericFileBrowser.xaml.cs:443]: @rjnb50 You can use `GetDoubleClickTime()` from Vanara library

**Comment by yaichenbaum** [comment]
@rjnb50 I am still getting the same issue as before after testing your changes...

**Comment by yaichenbaum** [comment]
@rjnb50 I am closing this pull request as we have not heard back from you, please reopen the pull request when you get a chance to review the comments.

**Pull request closed without merging**